broadinstitute/single_cell_portal_core
# Adding test for special characters in filenames (SCP-3364)

This update adds a test to validate that file uploads with special characters in the filenames (e.g. parentheses) still work.  This was found as an issue during the [Rails 6.1/Carrierwave](#1030) migration, and while it was addressed, a test was not added to directly validate this moving forward.  This test validates that all non-word characters are replaced with underscores (`_`).

TO TEST:
1. Boot portal as normal on this branch
2. In a study, navigate to the file upload wizard and select the 'Miscellaneous' tab
3. Upload the test file `test/test_data/README (1).txt`
4. Note that the file uploads w/o issue and is renamed to `README__1_.txt`

This PR satisfies SCP-3364.
